frontend/src/components/Chat.tsx

```diff
@@ -30,15 +33,44 @@ export function Chat() {
         crop?: string;
         files?: number;
     }>({});
+    const [showTooltip, setShowTooltip] = useState(false);
+    const { isAuthenticated, logout } = useAuth();
 
-    const handlePromptSubmit = (value: string) => {
-        console.log('Prompt submitted:', value);
-        // Handle prompt submission
-    };
+    // Chat store
+    const { status, events, userQuery, startStreaming, addEvent, finishStreaming, setError } = useChatStore();
+
+    const handlePromptSubmit = useCallback(async (value: string) => {
+        if (!value.trim()) return;
+
+        startStreaming(value);
+
+        const controller = createStreamController();
+
+        try {
+            await sendChatMessage(value, {
+                onMessage: (message) => {
+                    addEvent({
+                        ...message,
+                        id: `msg-${Date.now()}-${Math.random()}`,
```

Comment:
Do we want to do anything more interesting with message ids? E.g. I could create and send a message id (probably a uuid) server-side?